dashboard/src/components/AppView/AppControls/StatusAwareButton.tsx

```diff
@@ -0,0 +1,32 @@
+import { CdsButton } from "@cds/react/button";
+import ReactTooltip from "react-tooltip";
+import { hapi } from "../../../shared/hapi/release";
+
+export interface IStatusAwareButtonProps {
+  id: string;
+  releaseStatus: hapi.release.IStatus | undefined | null;
+}
+
+export default function StatusAwareButton<T extends IStatusAwareButtonProps>(props: T) {
+  const { id, releaseStatus, ...otherProps } = props;
+  const disabled: boolean = releaseStatus?.code
+    ? releaseStatus.code >= 5 && releaseStatus.code <= 8
+    : false;
+  const tooltips = {
+    5: "The applicatixon is being deleted.",
+    6: "The application is pending installation.",
+    7: "The application is pending upgrade.",
+    8: "The application is pending rollback.",
+  };
```

Comment:
I'd suggest extracting this logic to a separate function instead as it is really coupled to the helm status codes (of course, it should be aware of that, but I'd rather encapsulate it). See what you think.

Reply:
That is the exact purpose of the StatusAwareButton. I don't see a need to add one more layer of indirection.
All the button does is encapsulate the logic of computing the disabled flag and tooltip.
what would be the reason for another indirection?

Reply:
Ok, sounds reasonable. I was thinking about what if Helm changes their codes. But, the condition is rather simple, so there is no need to adding an extra layer as you mention.